jerry-core/ecma/builtin-objects/ecma-builtin-string-prototype.c

```diff
@@ -525,19 +623,46 @@ ecma_builtin_string_prototype_object_replace (ecma_value_t this_value, /**< this
         else
         {
           replace_ctx.matched_p = curr_p;
-          replace_ctx.match_byte_pos = byte_offset;
+          replace_ctx.match_byte_pos = replace_all ? (lit_utf8_size_t) (curr_p - replace_ctx.string_p) : byte_offset;
```

Comment:
Remove the condition, always assign `(lit_utf8_size_t) (curr_p - replace_ctx.string_p)`